common/app/layout/FaciaCard.scala

```diff
@@ -0,0 +1,259 @@
+    case Some(InlineVideo(_, _, Some(_))) => true
+    case Some(InlineYouTubeMediaAtom(_, _)) => true
+    case Some(InlineImage(_)) => true
+    case Some(InlineSlideshow(_)) => true
+    case Some(CrosswordSvg(_)) => true
+    case _ => false
+  }
+
+  def dataLinkName(index: Int): String = {
+    val name = s"$analyticsPrefix | card-@${index + 1}"
+    val withFromShowMore = if (fromShowMore) s"showmore | $name" else name
+    withFromShowMore
+  }
+
+  def withTimeStamp: ContentCard = copy(timeStampDisplay = Some(DateOrTimeAgo))
+
+  def showDisplayElement: Boolean =
+    cardTypes.allTypes.exists(_.canShowMedia) && !displaySettings.imageHide && cutOut.isEmpty
+
+  def showReviewStars: Boolean = starRating.isDefined && (displayElement match {
+    case Some(InlineImage(_)) if showDisplayElement => false
+    case _ => true
+  })
+
+  def showStandfirst: Boolean = cardTypes.allTypes.exists(_.showStandfirst)
+
+  def mediaWidthsByBreakpoint(implicit requestHeader: RequestHeader) : WidthsByBreakpoint =
+      FaciaWidths.mediaFromItemClasses(cardTypes)
+
+  def squareImageWidthsByBreakpoint(implicit requestHeader: RequestHeader) : WidthsByBreakpoint =
+      FaciaWidths.squareFront()
+
+  def showTimestamp: Boolean = timeStampDisplay.isDefined && webPublicationDate.isDefined
+
+  val analyticsPrefix = s"${cardStyle.toneString} | group-$group${if(displaySettings.isBoosted) "+" else ""}"
+
+  val hasInlineSnapHtml = snapStuff.exists(_.embedHtml.isDefined)
+
+  val isMediaLink = mediaType.nonEmpty
+
+  val hasVideoMainMedia = displayElement match {
+    case Some(_: InlineVideo) if !isMediaLink => true
+    case Some(_: InlineYouTubeMediaAtom) if !isMediaLink => true
+    case _ => false
+  }
+
+  val designType: Option[DesignType] = storyContent.map(_.metadata.designType)
+  val pillar: Option[Pillar] = Pillar(storyContent)
+  val contentType: DotcomContentType = DotcomContentType(storyContent)
+
+  val isAdvertisementFeature : Boolean = designType.contains(AdvertisementFeature)
+}
+
+object ContentCard {
+
+  def fromApiContent(apiContent: contentapi.Content): Option[ContentCard] = {
+
+    val cardTypesForRecommendations = ItemClasses(mobile = MediaList, tablet = Standard)
+
+    PartialFunction.condOpt(FaciaCard.fromTrail(
+      faciaContent = FaciaContentConvert.contentToFaciaContent(apiContent),
+      config = CollectionConfig.empty,
+      cardTypes = cardTypesForRecommendations,
+      showSeriesAndBlogKickers = false
+    )) {
+      case content: ContentCard => content
+    }
+
+  }
+}
+
+case class HtmlBlob(html: Html, customCssClasses: Seq[String], cardTypes: ItemClasses) extends FaciaCard
+
+case class PaidCard(
+  icon: Option[String],
+  headline: String,
+  kicker: Option[String],
+  description: Option[String],
+  image: Option[ImageMedia],
+  fallbackImageUrl: Option[String],
+  targetUrl: String,
+  cardTypes: Option[ItemClasses] = None,
+  branding: Option[Branding]
+) extends FaciaCard
```

Comment:
Hi @shtukas ! We're doing the Scala 2.13 upgrade in https://github.com/guardian/frontend/pull/25190, and the match-exhaustiveness check has pointed out that `PaidCard` is not handled in this pattern-match expression in the template that renders cards on regular (non-commercial?) fronts:

https://github.com/guardian/frontend/blob/95f470aaa275f0b204d157afca08952b799637b6/common/app/views/fragments/items/facia_cards/item.scala.html#L31-L56

....we were wondering what we should do to satisfy the compiler here? Have a blank case for `PaidCard`? There's already some logic around `paidContentOnEditorialPage` that seems to be in the same space?

Reply:
My instinct would be to maintain the existing behaviour and add a `case` for `PaidCard`s but without rendering anything.